File: backend/src/main/java/com/example/backend/service/BookingService.java
package com.example.backend.service;

import com.example.backend.model.Booking;
import com.example.backend.model.dto.BookingDto;
import com.example.backend.model.forms.BookingForm;

import java.util.List;
import java.util.Optional;

public interface BookingService {
    Optional<Booking> addNewBooking(BookingForm bookingForm);
    List<BookingDto> getUpcomingBookingsForUser(String username);
    List<BookingDto> getAllBookingsForUser(String username);
    Optional<Booking> cancelBookingForTrip(String bookingId);
    List<BookingDto> getAllBookingsForTrip(String tripId);
    Optional<Booking> payForBookingForTrip(String bookingId);
    Optional<BookingDto> getBookingDetails(String bookingId);
}

File: frontend/src/services/BookingService.js
import axios from "../custom-axios/axiosInstance";

const BookingService={
    getUpcomingBookingsForUser:(username)=>{
        return axios.get(`/api/bookings/upcomingBookings/${username}`)
    },

    getAllBookingsForUser:(username)=>{
        return axios.get(`/api/bookings/allBookings/${username}`)
    },

    getBookingDetails:(bookingId)=>{
        return axios.get(`/api/bookings/details/${bookingId}`)
    },

    payForBookingForTrip:(bookingId)=>{
        return axios.get(`/api/bookings/payForTrip/${bookingId}`)
    },

    cancelBookingForTrip:(bookingId)=>{
        return axios.get(`/api/bookings/cancelBooking/${bookingId}`)
    },

    getAllBookingsForTrip:(tripId)=>{
        return axios.get(`/api/bookings/forTrip/${tripId}`)
    },

    addNewBooking:(username,tripId, roomId, totalPrice)=>{
        return axios.post(`/api/bookings/addNewBooking`,{
            "username": username,
            "tripId": tripId,
            "roomId": roomId,
            "totalPrice": totalPrice,
        })
    }
}

export default BookingService;